flaskulator.py

```diff
@@ -0,0 +1,81 @@
+from flask import Flask, render_template, request
+from decimal import Decimal
+
+app = Flask(__name__)
+app.config.from_object(__name__)
+
+# define user-defined exceptions
+class Error(Exception):
+    """Base class for other exceptions"""
+    pass
+
+class NoNumberError(Error):
+    """Raised when one of the number fields is empty"""
+    pass
+
+class ZeroDivError(Error):
+    """Raised when division by zero is executed"""
+    pass
+
+@app.route('/')
+def home_page():
+    return render_template("index.html")
+
+@app.route('/result', methods=['GET', 'POST'])
+def result():
+    operator = {
+        "+": add,
+        "-": subtract,
+        "*": mult,
+        "/": divide
+    }
+    form = request.form
+    number1 = request.form['number1']
+    number2 = request.form['number2']
+    operation = request.form['operation']
+    try:
+        # check if the number fields are not empty
+        if len(number1) < 1 or len(number2) < 1:
+            raise NoNumberError
+        # zero division checking
+        if operation == "/" and number2 == "0":
+            raise ZeroDivError
+        #result calculation
+        result = operator[operation](Decimal(number1), Decimal(number2))
+        #Result representation.
+        if len(str(result)) > 5:
+            result = "%.4g" % result # set result precision to 4 digits
+        return render_template("result.html", result=result, form=form)
+    except ZeroDivError:
+        error = "ZeroDivError"
+        result = "Infinity (division by zero)"
+        return render_template("result.html", result=result, form=form, error=error)
```

Comment:
in your result template I saw, that firstly you check { if error }. If there is an error, else block doesn't run. For what reason in this section do you pass an "result" and "form"?